benchs/gemm/bench.py

```diff
@@ -0,0 +1,274 @@
+import torch
+from torch import Tensor
+from typing import Tuple
+import sys
+import os
+import csv
+
+cutlass_dir = os.path.join(os.path.dirname(__file__), 'cutlass')
+sys.path.insert(0, cutlass_dir)
+
+tiledcuda_dir = os.path.join(os.path.dirname(__file__), 'tiledcuda')
+sys.path.insert(0, tiledcuda_dir)
+
+from cutlass.gemm import gemm_func as cutlass_gemm
+from tiledcuda.gemm import gemm_func as tiledcuda_gemm
+from cuBLAS import cublas_gemm
+
+def run_cublas_unittest(
+        a: Tensor,
+        b: Tensor,
+        c: Tensor,
+        M: int,
+        N: int,
+        K: int,
+        debug_print=False,
+        epsilon: float = 5e-2
+):
+    time = torch.zeros(1, device=torch.device("cpu"), dtype=torch.float32)
+    cublas_gemm(M, N, K, a, b, c, time)
+    ref_c = a @ b.t()
+
+    if debug_print:
+        print("Result:")
+        print(c)
+
+        print("\nReference:")
+        print(ref_c)
+
+    avg_diff = (torch.sum(torch.abs(ref_c - c) / (M * N))).item()
+
+    if avg_diff > epsilon:
+        return False
+    else:
+        return True
+
+def run_tiledcuda_unittest(
+        a: Tensor,
+        b: Tensor,
+        c: Tensor,
+        M: int,
+        N: int,
+        K: int,
+        kTM: int,
+        kTN: int,
+        kTK: int,
+        kRK: int,
+        warp_layout: Tuple,
+        debug_print=False,
+        epsilon: float = 5e-2
+):
+    tiledcuda_gemm(a, b, c, M, N, K, kTM, kTN, kTK, kRK, *warp_layout)
+    ref_c = a @ b.t()
+
+    if debug_print:
+        print("Result:")
+        print(c)
+
+        print("\nReference:")
+        print(ref_c)
+
+    avg_diff = (torch.sum(torch.abs(ref_c - c.half()) / (M * N))).item()
+
+    if avg_diff > epsilon:
+        return False
+    else:
+        return True
+
+def run_cutlass_unittest(
+        a: Tensor,
+        b: Tensor,
+        c: Tensor,
+        M: int,
+        N: int,
+        K: int,
+        kTM: int,
+        kTN: int,
+        kTK: int,
+        warp_layout: Tuple,
+        debug_print=False,
+        epsilon: float = 5e-2
+):
+    cutlass_gemm(a, b, c, M, N, K, kTM, kTN, kTK, *warp_layout)
+    ref_c = a @ b.t()
+
+    if debug_print:
+        print("Result:")
+        print(c)
+
+        print("\nReference:")
+        print(ref_c)
+
+    avg_diff = (torch.sum(torch.abs(ref_c - c) / (M * N))).item()
+
+    if avg_diff > epsilon:
+        return False
+    else:
+        return True
+
+def run_cublas_bench(
+    a: Tensor,
+    b: Tensor,
+    c: Tensor,
+    M: int,
+    N: int,
+    K: int,
+    time: Tensor
+):
+    # Run unittest
+    if run_cublas_unittest(a, b, c, M, N, K):
+        # print("Unittest passed")
+        pass
+    else:
+        # raise ValueError("Unittest failed")
+        print("Run cuBLAS unittest failed")
```

Comment:
Do you still want to keep lines 120 and 123?

Reply:
fixed.